src/main/java/roomescape/controller/advice/GlobalAdvice.java

```diff
@@ -0,0 +1,63 @@
+package roomescape.controller.advice;
+
+import java.util.List;
+import org.springframework.context.MessageSourceResolvable;
+import org.springframework.context.support.DefaultMessageSourceResolvable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.method.annotation.HandlerMethodValidationException;
+import roomescape.exception.BadRequestException;
+import roomescape.exception.NotFoundException;
+
+@ControllerAdvice
+public class GlobalAdvice {
+
+    @ExceptionHandler(NotFoundException.class)
+    public ResponseEntity<String> notFoundExceptionHandler(NotFoundException exception) {
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(exception.getMessage());
+    }
+
+    @ExceptionHandler(BadRequestException.class)
+    public ResponseEntity<String> badRequestExceptionHandler(BadRequestException exception) {
+        return ResponseEntity.badRequest().body(exception.getMessage());
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<String> methodArgumentNotValidExceptionHandler(MethodArgumentNotValidException exception) {
+        List<FieldError> fieldErrors = exception.getBindingResult().getFieldErrors();
+        List<String> messages = fieldErrors.stream()
+                .map(DefaultMessageSourceResolvable::getDefaultMessage)
+                .toList();
+        return ResponseEntity.badRequest().body(String.join("\n", messages));
+    }
+
+    @ExceptionHandler(HandlerMethodValidationException.class)
+    public ResponseEntity<String> handlerMethodValidationExceptionHandler(HandlerMethodValidationException exception) {
+        List<String> errorMessage = exception.getAllErrors().stream()
+                .map(MessageSourceResolvable::getDefaultMessage)
+                .toList();
+        return ResponseEntity.badRequest().body(String.join("\n", errorMessage));
+    }
+
+    @ExceptionHandler(HttpMessageNotReadableException.class)
+    public ResponseEntity<String> httpMessageNotReadableExceptionHandler(HttpMessageNotReadableException exception) {
+        return ResponseEntity.badRequest().body("형식이 적절하지 않은 요청메세지 입니다.");
+    }
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<String> illegalArgumentExceptionHandler(IllegalArgumentException exception) {
+        System.out.println("exception = " + exception);
+        return ResponseEntity.internalServerError().body("서버 내부에서 로직 예외 발생헸습니다.");
```

Comment:
IllegalArgumentException 은 보통 400에러로 처리합니다.
아래와 같이 사용자의 입력이 잘못되었고, 어떤 부분이 잘못되었는지 명시적으로 나타내주는 것이 좋겠네요!

```suggestion
        return ResponseEntity.badRequest().body(exception.getMessage());
```

Reply:
앗.. IllegalArgumentException는 당연히 서버에러라고 관성적으로 생각하고 있었는데 이번 기회를 통해 400으로 처리한다는 점을 처음 배우게 되었습니다. 

그리고 제가 관성적으로 IllegalArgumentException을 사용하고 있었다는 점과 IllegalArgumentException을 서버측에서 발생하는 일반적인 예외처럼 의도에 맞지않게 사용했다는 점을 반성할 수 있었습니다.

이러한 점에서 추천해주신 방식도 좋지만 IllegalArgumentException 처리한 부분을 InternalServerException이라는 커스텀 예외로 바꿔서 핸들링을 하는 편이 예외상황에 더욱 적절한 처리라는 판단이 들었습니다. 혹시 잘못된 점을 짚어주신다면 다시 곰곰이 고민해보고 수정해보겠습니다!